3-make_gene_matrix.R

```diff
@@ -0,0 +1,122 @@
+        help = "GEO ID of dataset for downloading the metadata",
+        metavar = "character"),
+    make_option(opt_str = c("-o", "--output"), type = "character",
+        default = getwd(), help = "Directory where you would like the output to go",
+        metavar = "character"),
+    make_option(opt_str = c("-m", "--mapped"), type = "numeric", 
+        default = "0.5", help = "Cutoff for what percent mapped_reads samples
+        should have. Any samples with less than the cutoff will be removed.",
+        metavar = "numeric"), 
+    make_option(opt_str = c("-l", "--label"), type = "character",
+        default = "", help = "Optional label for output files",
+        metavar = "character")
+)
+
+# Parse options
+opt <- parse_args(OptionParser(option_list = option_list))
+
+#------------------------------Import Salmon reads-----------------------------#
+# Get quant files
+quant.files <- list.files(opt$dir, recursive = TRUE, full.names = TRUE,
+                       pattern = "quant.sf")
+
+# Carry sample names over
+sample.names <- stringr::word(quant.files, 3, sep = "/")
+
+# Get transcript IDs
+transcripts <- read.table(quant.files[[1]], header = TRUE, 
+                 colClasses = c("character", rep("NULL", 4)))
+
+transcripts.wout.ver <- gsub("\\.[0-9]*$", "", transcripts$Name)
+
+# Turn transcript ids to gene ids
+tx.gene.key <- data.frame('transcript' = transcripts$Name,
+                          'gene' = mapIds(org.Hs.eg.db, keys = transcripts.wout.ver,
+                                         column = "ENSEMBL", keytype = "ENSEMBLTRANS"),
+                          stringsAsFactors = FALSE)
+
+# Remove transcripts without genes
+tx.gene.key <- tx.gene.key[!is.na(tx.gene.key$gene),]
+
+# Do the thing
+tx.counts <- tximport::tximport(quant.files, type = "salmon", tx2gene = tx.gene.key,
+                                countsFromAbundance = "no")
+
+# make as a dataframe
+tx.counts <- data.frame(tx.counts$counts, stringsAsFactors = FALSE)
+
+# Get gene symbols for data
+gene <- mapIds(org.Hs.eg.db, keys = tx.gene.key$gene , column = "SYMBOL",
+               keytype = "ENSEMBL")
+
+#---------------------Salmon proportion of mapped reads------------------------#
+# Get the proportion of mapped reads by reading the meta files
+salmon.prop.assigned <- vapply(sample.names, function(x) {
+  rjson::fromJSON(file = file.path(opt$dir, x, "aux_info",
+                                   "meta_info.json"))$percent_mapped/100
+  }, FUN.VALUE = 1)
+
+# Make a histogram of this information
+png(file.path(opt$output, paste0(opt$label, "_prop_reads_mapped_hist.png")))
+hist(salmon.prop.assigned, xlab = "", main = "Proportion of Mapped Reads",
+breaks = 20)
+dev.off()
+
+# Filter out samples with too low of mapped reads
+tx.counts <- tx.counts[, which(salmon.prop.assigned > opt$mapped)]
+
+# Save to an RDS file
+saveRDS(salmon.data, file = file.path(opt$output, 
+                                      paste0(opt$label, ".counts.data.RDS")))
```

Comment:
I think you want to go TSV or CSV here. Again, you can (most likely) write a compressed file if you're concerned about size.